gemrb/core/GUI/GameControl.cpp

```diff
@@ -59,6 +59,13 @@ static const ResRef TestSpell = "SPWI207";
 
 uint32_t GameControl::DebugFlags = 0;
 
+static Actor* GetMainActor()
+{
+	if (auto actor = core->GetFirstSelectedPC(false))
+		return actor;
+	return core->GetGame()->selected[0];
+}
+
```

Comment:
I don't think we need this function. We can use `game->selected[0]` directly or more nicely, through `Interface::GetFirstSelectedActor()`. One of the places already ensures it's set, while the others can get asserts, since they should have it too.

Reply:
the confusion (for me at least) is that `Interface::GetFirstSelectedPC` is doing something other than returning `selected[0]` and all the code in `GameControl` was following this pattern of try `Interface::GetFirstSelectedPC` and if that returns null then use `selected[0]`.

I dont believe we can just use `GetFirstSelectedActor`/`selected[0]` because that could be an NPC and the code calling this looks like it purposely wants to prioritize a PC over NPC. Wether it _should_ do that or not I do not know.

Reply:
It's using `selected` like that for historic reasons. If there's only one selected thing, then it doesn't matter. If there's several, most of the special actions aren't available — only attack and defend are, which are both for everyone. Ergo using the first thing in the list is just fine.

Reply:
I'm confused. It still sounds like if you have multiple actors selected then `selected[0]` still may not be a PC so in this example you could have a group selected where the first actor is an NPC then direct the group to open a container and that NPC would then be directed to the container rather than a PC. Is that not accurate?

Reply:
Ah, that's true, the odd one out. We could ensure the pcs are on top when manipulating the list, but so far it seems cleaner to have another lookup method. Then I have other comments.

Reply:
should i just delete the function and make calls for GetFirstSelectedPC instead? if core->GetGame()->selected[0] can return some summoned unit or whatever, that's not what we need right here

Reply:
But there might also be no pc selected, so that is not enough. As far as inlining back, I think it should stay. The function might come in handy some place else, but more importantly, since it's more thorough, it makes reasoning about the code easier, since we can't just use `selected[0]` everywhere.

Reply:
done